examples/siro_sandbox/gui_pick_helper.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import magnum as mn
+import numpy as np
+
+
+class GuiPickHelper:
+    def __init__(self, gui_service, agent_idx):
+        self._sandbox_service = gui_service
+        self._agent_idx = agent_idx
+        self._rom = self._get_sim().get_rigid_object_manager()
+        self.obj_ids = self._get_sim()._scene_obj_ids
+
+    def _get_sim(self):
+        return self._sandbox_service.sim
+
+    def _closest_point_and_dist(self, origin, direction_vector, points):
+        norm_direction = direction_vector / np.linalg.norm(direction_vector)
+        vectors_to_points = points - origin
+        dot_products = np.dot(vectors_to_points, norm_direction)
+        closest_points = origin + dot_products[:, np.newaxis] * norm_direction
+        distances = np.linalg.norm(closest_points - points, axis=1)
+        return np.argmin(distances), np.min(distances)
+
+    def on_environment_reset(self):
+        sim = self._get_sim()
+        self._rom = sim.get_rigid_object_manager()
+        self.obj_ids = sim._scene_obj_ids
+        self.agent_feet_height = 0.15
```

Comment:
Can we avoid this magic number? Can you set it from the appropriate data?

Reply:
Can you also use the `self._member` naming convention for protected members (those that won't be used outside this class)?